pandas/core/frame.py

```diff
@@ -4473,6 +4473,10 @@ def query(self, expr: str, *, inplace: bool = False, **kwargs) -> DataFrame | No
         """
         Query the columns of a DataFrame with a boolean expression.
 
+        This function pass the `expr` parameter to :meth:`~pandas.DataFrame.eval`.
+        This allows `eval` to run arbitrary code, which can make you vulnerable to code
+        injection if you pass user input to this function.
```

Comment:
```suggestion
        This method can run arbitrary code, which can make you vulnerable to code
        injection if you pass user input to this function.
```

I don't think we need to mention the implementation details; it's sufficient to warn that query can run arbitrary code.